src/Message/AbstractRequest.php

```diff
@@ -1139,17 +1151,14 @@ protected function buildPaymentMethod($paymentMethodType, $addAttributes = false
             if (!$this->isUpdate()) {
                 $paymentMethod->type = self::PAYMENT_METHOD_PAYPAL;
             }
-        } elseif ($paymentMethodType === self::PAYMENT_METHOD_APPLE_PAY && $card !== null) {
+        } elseif ($paymentMethodType === self::PAYMENT_METHOD_APPLE_PAY) {
             $applePay = new stdClass();
 
-            /**
-             * @var \Omnipay\Vindicia\NonStrippingCreditCard $card
-             */
-            $applePay->paymentInstrumentName = $card->getPaymentInstrumentName();
-            $applePay->paymentNetwork = $card->getPaymentNetwork();
-            $applePay->paymentData = $card->getToken();
-            $applePay->transactionReference = $card->getApplePayTransactionReference();
-            $applePay->expirationDate = $card->getExpiryDate(self::VINDICIA_EXPIRATION_DATE_FORMAT);
+            $token = $this->getApplePayToken();
+            $applePay->paymentInstrumentName = $token['paymentMethod']['displayName'];
```

Comment:
It will never return null because of this line [here](https://github.com/vimeo/omnipay-vindicia/pull/59/files#diff-5e5bf1918fda01730b8630e60acf105eR88), which makes sure that that param is set before continuing.

Reply:
Ok could you add a comment explaining the validation that is done before we access `getApplePayToken()` to ensure that it will never be null at this point?